### PR TITLE
fix: add MariaDB support for unique indexes

### PR DESCRIPTION
**Pull request by freestingo**
### What does it do?

Activating migrations and editing schemas (i.e. adding a new field) will no longer throw an error when starting the application using MariaDB. 

### Why is it needed?

The bug stems from a wrong assumption about the `Non_unique` field type returned by the DB schema; it was expected to be a `boolean` value, while MariaDB just returns either a `'0'` or a `'1'` string. The logic now handles both cases, and therefore fixes the bug.

### Related issue(s)/PR(s)

All additional related info can be found in #16959 .

### CODE REVIEW

File: packages/core/database/src/dialects/mysql/schema-inspector.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ interface RawColumn {
 interface RawIndex {
   Key_name: string;
   Column_name: string;
-  Non_unique: boolean;
+  Non_unique: boolean | string;
```
Comment (innerdvations) [inline packages/core/database/src/dialects/mysql/schema-inspector.ts:22]:
```suggestion
  Non_unique: boolean | '0' | '1' ;
```

**Comment by alexandrebodin** [comment]
> [!IMPORTANT] 
> With the release of Strapi 5, we have lifted the code freeze. To ensure we can effectively review and assist with contributions, we are now only accepting PRs targeting version 5 (the develop branch).<br/> 
> As your PR is targeting some of the most recent v4 version, We we would greatly appreciate it if you could update your contribution to align with version 5 and resubmit!

**Comment by innerdvations** [comment]
Agreed, I'll add it to our backlog and update it ourselves if we need to.

**Review by innerdvations — APPROVED** [review]
Tested and works for me!

**Comment by alexandrebodin** [comment]
@freestingo Thank you for your contribution 💯

**Pull request merged**